build.gradle.kts

```diff
@@ -5,323 +5,325 @@ import kotlin.collections.ArrayList
+  classifier = "src"
 
-    from(sourceSets.main.get().allSource)
+  from(sourceSets.main.get().allSource)
 }
 
 extra.apply {
-   val df = SimpleDateFormat("yyyy")
-   val year = df.format(Date())
-   val javaHome = System.getProperty("java.home") ?: throw GradleException("java.home is not set")
-   val cmd = javaHome + File.separator + "bin" + File.separator + "jpackage"
-   val jpackagecmd = if (cmd.contains(" ")) "\"" + cmd + "\"" else cmd
-   val parameters = ArrayList<String>(Arrays.asList(
-      jpackagecmd,
+  val df = SimpleDateFormat("yyyy")
+  val year = df.format(Date())
+  val javaHome = System.getProperty("java.home") ?: throw GradleException("java.home is not set")
+  val cmd = javaHome + File.separator + "bin" + File.separator + "jpackage"
+  val jPackageCmd = if (cmd.contains(" ")) "\"" + cmd + "\"" else cmd
+  val parameters = ArrayList<String>(Arrays.asList(
+      jPackageCmd,
       "--input", "$buildDir/libs",
       "--main-class", "com.cburch.logisim.Main",
       "--main-jar", project.name + '-' + project.version + "-all.jar",
       "--app-version", project.version as String,
       "--copyright", "Copyright © 2001–" + year + " Carl Burch, BFH, HEIG-VD, HEPIA, Holy Cross, et al.",
       "--dest", "$buildDir/dist"
-   ))
-   val linuxParameters = ArrayList<String>(Arrays.asList(
+  ))
+  val linuxParameters = ArrayList<String>(Arrays.asList(
       "--name", project.name,
       "--file-associations", "$projectDir/support/jpackage/linux/file.jpackage",
       "--icon", "$projectDir/support/jpackage/linux/logisim-icon-128.png",
       "--install-dir", "/opt",
       "--linux-shortcut"
-   ))
-   set("sharedParameters", parameters)
-   set("linuxParameters", linuxParameters)
-   set("jpackagecmd", jpackagecmd)
-   val projectName = project.name as String
-   val projectVersion = project.version as String
-   val uppercaseProjectName = projectName.substring(0,1).toUpperCase() + projectName.substring(1)
-   set("uppercaseProjectName", uppercaseProjectName)
-   set("appDirname", "$buildDir/dist/" + projectName + ".app")
-   set("dmgFilename", "$buildDir/dist/" + projectName + "-" + projectVersion + ".dmg")
-   set("rpmFilename", "$buildDir/dist/" + projectName + "-" + projectVersion + "-1.x86_64.rpm")
-   set("debFilename", "$buildDir/dist/" + projectName + "-" + projectVersion + "-1_amd64.deb")
-   set("msiFilename", "$buildDir/dist/" + projectName + "-" + projectVersion + ".msi")
+  ))
+  set("sharedParameters", parameters)
+  set("linuxParameters", linuxParameters)
+  set("jPackageCmd", jPackageCmd)
+  val projectName = project.name as String
+  val projectVersion = project.version as String
+  val uppercaseProjectName = projectName.substring(0,1).toUpperCase() + projectName.substring(1)
+  set("uppercaseProjectName", uppercaseProjectName)
+  set("appDirname", "$buildDir/dist/" + uppercaseProjectName + ".app")
+  set("dmgFilename", "$buildDir/dist/" + projectName + "-" + projectVersion + ".dmg")
```

Comment:
Just a question (not a change request really) - why some parts use concatenation (as this one) while other use just regular string interpolation?

Reply:
The reason is that I never considered using the $localVar form for local variables. That does not mean I thought it was a bad idea. I literally never thought about it at all. Now that I have, thanks to you, I may change it. I'll look at various ways to write it to see which seems to be easiest to read.

Reply:
Hint: Kotlin supports `${var}` syntax as well, and I personally always favor that form as it makes variable less "blended" with the string. i.e.

    set("dmgFilename", "${buildDir}/dist/${projectName}-${projectVersion}.dmg")

before

    set("dmgFilename", "$buildDir/dist/" + projectName + "-" + projectVersion + ".dmg")

Reply:
I have made these changes.